Parser fix

Abdulraheem-Arar opened the pull request:
I added the changes to the parser that I made onto this new branch and I am creating a pull request.

Abdulraheem-Arar committed:
Update SIL.scala

added the changes for the debug step and begin borrow parsing operations

Abdulraheem-Arar committed:
Update SILParser.scala

changes for debug_step and begin_ borrow for this file in order to parse them correctly

Abdulraheem-Arar committed:
Update SILPrinter.scala

made changes here for begin borrow and debug step parsing

Abdulraheem-Arar committed:
Update SWIRLGen.scala

updated to account for the debug step statement

Abdulraheem-Arar committed:
Update build.gradle

trying to see if the tests will pass when using this version of the api as taint analysis seems to not be working anymore while command execution is fine but the functionality of taint analysis is gone.

Abdulraheem-Arar committed:
Update SILParser.scala

fixed issue in begin_borrow

Abdulraheem-Arar committed:
Update build.gradle

Abdulraheem-Arar committed:
Update ubuntu.yml

Abdulraheem-Arar committed:
Update ubuntu.yml

Abdulraheem-Arar committed:
Update build.gradle

karimhamdanali committed:
add support for move_value instruction + update alloc_stack handling

NinjaLikesCheez commented:
I opened https://github.com/themaplelab/swan/pull/92 pointing to this branch to include another fix for a debug attribute - if you'd prefer I hold off and merge it to `spds` after this merges let me know! :)

karimhamdanali committed:
change SPDS version back to 3.1.1

This change includes the following:
- changing the version in build.gradle
- higher priority to local maven libraries (to be resolved before maven central)
- adding pre-built libraries for 3.1.1, which I have compiled from source

karimhamdanali committed:
fix typo in comments

karimhamdanali committed:
fix SWAN CLI build scripts

- specify particular version for library dependcies for swan-swiftc and swan-xcodebuild
- use proper executableTarget in Package.swift

karimhamdanali committed:
update README with specific Xcode and Swift versions that we have recently tested SWAN on
